test/unit/BPool.t.sol

```diff
@@ -725,21 +726,124 @@ contract BPool_Unit_Finalize is BasePoolTest {
+    _setTotalWeight(_fuzz.totalWeight);
+  }
 
-  function test_Set_TokenArray() private view {}
+  function _assumeHappyPath(Bind_FuzzScenario memory _fuzz) internal {
+    vm.assume(_fuzz.token != VM_ADDRESS);
+    vm.assume(_fuzz.token != 0x000000000000000000636F6e736F6c652e6c6f67);
+    vm.assume(_fuzz.balance >= MIN_BALANCE);
+    vm.assume(_fuzz.totalWeight >= MIN_WEIGHT);
+    vm.assume(_fuzz.totalWeight <= MAX_TOTAL_WEIGHT - MIN_WEIGHT);
+    _fuzz.previousTokensAmount = bound(_fuzz.previousTokensAmount, 0, MAX_BOUND_TOKENS - 1);
+    _fuzz.denorm = bound(_fuzz.denorm, MIN_WEIGHT, MAX_TOTAL_WEIGHT - _fuzz.totalWeight);
+    _fuzz.previousTokens = new address[](_fuzz.previousTokensAmount);
+    for (uint256 i = 0; i < _fuzz.previousTokensAmount; i++) {
+      _fuzz.previousTokens[i] = makeAddr(i.toString());
+      vm.assume(_fuzz.token != _fuzz.previousTokens[i]);
+    }
+  }
 
-  function test_Emit_LogCall() private view {}
+  modifier happyPath(Bind_FuzzScenario memory _fuzz) {
+    _assumeHappyPath(_fuzz);
+    _setValues(_fuzz);
+    _;
+  }
+
+  function test_HappyPath(Bind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    bPool.bind(_fuzz.token, _fuzz.balance, _fuzz.denorm);
+  }
+
+  function test_Revert_NotController(
+    address _controller,
+    address _caller,
+    Bind_FuzzScenario memory _fuzz
+  ) public happyPath(_fuzz) {
+    vm.assume(_controller != _caller);
+    bPool.set__controller(_controller);
+
+    vm.prank(_caller);
+    vm.expectRevert('ERR_NOT_CONTROLLER');
+    bPool.bind(_fuzz.token, _fuzz.balance, _fuzz.denorm);
+  }
+
+  function test_Revert_IsBound(uint256 _tokenIndex, Bind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    vm.assume(_fuzz.previousTokensAmount > 0);
+    _tokenIndex = bound(_tokenIndex, 0, _fuzz.previousTokens.length - 1);
+
+    vm.expectRevert('ERR_IS_BOUND');
+    bPool.bind(_fuzz.previousTokens[_tokenIndex], _fuzz.balance, _fuzz.denorm);
+  }
+
+  function test_Revert_Finalized(Bind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    _setFinalize(true);
+
+    vm.expectRevert('ERR_IS_FINALIZED');
+    bPool.bind(_fuzz.token, _fuzz.balance, _fuzz.denorm);
+  }
+
+  function test_Revert_MaxPoolTokens(Bind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    address[] memory _tokens = _setRandomTokens(MAX_BOUND_TOKENS);
+    for (uint256 i = 0; i < _tokens.length; i++) {
+      vm.assume(_fuzz.token != _tokens[i]);
+    }
+
+    vm.expectRevert('ERR_MAX_TOKENS');
+    bPool.bind(_fuzz.token, _fuzz.balance, _fuzz.denorm);
+  }
 
-  function test_Call_Rebind() private view {}
+  function test_Set_Record(Bind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    bPool.bind(_fuzz.token, _fuzz.balance, _fuzz.denorm);
+
+    assertTrue(bPool.isBound(_fuzz.token));
+    assertEq(bPool.call__records(_fuzz.token).index, _fuzz.previousTokensAmount);
+  }
+
+  function test_Set_TokenArray(Bind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
```

Comment:
Following https://github.com/defi-wonderland/balancer-v1-amm/pull/45#discussion_r1619917008, `Push` is more explicit than `Set` here.

Reply:
I'm not sure about that change, we are using `Push` to describe when a ERC20 token is sent, and using it here can be confusing.